doc/sfp-cmis/cmis-init.md

```diff
@@ -0,0 +1,216 @@
+
+The feature is built on top of SONiC **sfp-refactor** framework to provide a platform-independent
+solution, and the individual platforms could easily enable this feaure by having its **Sfp** inherited from **SfpOptoeBase**.
+
+**Example:**  
+```
+from sonic_platform_base.sonic_xcvr.sfp_optoe_base import SfpOptoeBase
+
+class Sfp(SfpOptoeBase):
+
+    def __init__(self, sfp_index):
+        SfpOptoeBase.__init__(self)
+        self.index = sfp_index
+
+    def get_port_type(self):
+        return self.SFP_PORT_TYPE_QSFPDD
+
+    def get_eeprom_path(self):
+        # platform-specific per-port I2C bus
+        i2c_bus = 32 + self.index
+        return "/sys/bus/i2c/devices/{}-0050/eeprom".format(i2c_bus)
+```
+
+The scope of this feature is as follow:
+
+- **sonic-platform-common**  
+  - Enhance the **sonic-xcvr** to support CMIS application advertising and initialization
+- **sonic-platform-daemons**  
+  - Enhance the **sonic-xcvrd** to support paralleled CMIS application initialization
+- **sonic-utilities**  
+  - Enhance the **sfputil** and **sfpshow** to show CMIS application advertisement
+
+## Functional Requirements
+
+1. Ability to parse the advertised applications from the transceivers
+2. Ability to post the advertised applications to the STATE_DB
+3. Ability to support parallel processing of the application initialization
+4. Ability to detect the errors of the application initialization on the transceivers
+
+**Note:**  
+The duration of the CMIS application initialization greatly differs from transceivers
+to transceivers, while some take 3 seconds for activating the 4x100G mode, some require 15 seconds.
+Hence it's mandatory to support parallel processing.
+
+## Warm Boot Requirements
+
+Functionality should continue to work across warm boot. 
+- The CMIS application initialization should not be performed during WarmBoot.
+- The CMIS application initialization should be skipped if no application code updates.
+
+# Functional Description
+
+- The **pmon#xcvrd** will detect the module type of the attched transceivers and post the
+information to the **STATE_DB**
+- If the transceiver is a CMIS module, the **pmon#xcvrd** will activate the appropriate
+application mode as per the current port mode.
+- If the transceiver is a CMIS module, the **show interfaces transceiver eeprom**
+should display the advertised applications of the transceiver.
+- If the transceiver is a CMIS module, the **sfputil** should be capable of detecting the errors of the CMIS application initialization.
+
+# Design
+
+## sonic-platform-daemons/sonic-xcvrd (modified)
+
+The transceiver daemon will be enhanced as below:
+
+- Post the CMIS application advertisement into the **STATE_DB**  
+  In the case of CMIS transceiver, the **application_advertisement** of **TRANSCEIVER_INFO** is as follow:  
```

Comment:
"application_advertisement" is already part of the current DB schema:

```
admin@sonic:~$ redis-cli -n 6 hget "TRANSCEIVER_INFO|Ethernet0" "application_advertisement"
"400G CR8 - Copper cable   200GBASE-CR4 (Clause 136) - Copper cable   100GBASE-CR2 (Clause 136) - Copper cable  100GBASE-CR4 (Clause 92) - Copper cable   50GBASE-CR (Clause 126) - Copper cable   50GBASE-CR2 with no FEC - Copper cable   25GBASE-CR CA-N (Clause 110) - Copper cable   1000BASE -CX(Clause 39) - Copper cable   "
```

Reply:
Thanks for the info, I'll later rephrase this to 'Updating the format from plain test to the json dictionary format'. Hope it works for everyone

Reply:
BTW, the current cmis.py of sonic-xcvr does not support parsing application advertisement, may I ask how did you get this?

Reply:
QSFP-DD EEPROM parser is parsing EEPROM following CMIS, I guess it is also applied to other CMIS cables, so maybe you can implement cmis.py based on it.  https://github.com/Azure/sonic-platform-common/blob/master/sonic_platform_base/sonic_sfp/qsfp_dd.py#L101

Reply:
Actually, we did use it as the reference, both qsfp_dd.py and the current code PR use the ids defined in sff8024.py.
But the entire sonic-xcvr seems to be a rework, hence we will implement new parsers in the cmis.py of sonic-xcvr instead of the original one.